eth/sync.go

```diff
@@ -128,34 +129,30 @@ func (pm *ProtocolManager) txsyncLoop() {
 // syncer is responsible for periodically synchronising with the network, both
 // downloading hashes and blocks as well as handling the announcement handler.
 func (pm *ProtocolManager) syncer() {
+	// Start and ensure cleanup of sync mechanisms
 	pm.fetcher.Start()
 	defer pm.fetcher.Stop()
 	defer pm.downloader.Terminate()
 
-	sync := func() { pm.synchronise(pm.peers.BestPeer()) }
-	for {
-		batchTimer := time.AfterFunc(10*time.Second, sync)
-		for {
-			select {
-			case <-pm.noMorePeers:
-				batchTimer.Stop()
-				return
-
-			case <-pm.newPeerCh:
-				if pm.peers.Len() < minDesiredPeerCount {
-					continue
-				}
-
-				if batchTimer.Stop() {
-					go sync()
-				}
-
-			case <-batchTimer.C:
+	// Wait for different events to fire synchronisation operations
+	forceSync := time.NewTicker(forceSyncCycle)
+	defer forceSync.Stop()
 
+	for {
+		select {
+		case <-pm.newPeerCh:
+			// Make sure we have peers to select from, then sync
+			if pm.peers.Len() < minDesiredPeerCount {
+				break
 			}
-			// sync launched
+			go pm.synchronise(pm.peers.BestPeer())
 
-			break
+		case <-forceSync.C:
+			// Force a sync even if not enough peers are present
+			go pm.synchronise(pm.peers.BestPeer())
```

Comment:
It is harmless to call this, even though the downloader might already by syncing. Not sure if it warrants or should have a conditional on `if !pm.downloader.Synchronising()` or not. 

And but the `forceSync` thing is note worthy too.